sources/zarr/large_image_source_zarr/__init__.py

```diff
@@ -659,12 +677,70 @@ def crop(self, value):
             raise TileSourceError(msg)
         self._crop = (x, y, w, h)
 
+    def _generateDownsampledLevels(self, resample_method):
+        self._checkEditable()
+        current_arrays = dict(self._zarr.arrays())
+        if '0' not in current_arrays:
+            msg = 'No root data found, cannot generate lower resolution levels.'
+            raise TileSourceError(msg)
+        if 'x' not in self._axes or 'y' not in self._axes:
+            msg = 'Data must have an X axis and Y axis to generate lower resolution levels.'
+            raise TileSourceError(msg)
+
+        metadata = self.getMetadata()
+        tile_size = dict(width=4096, height=4096)
+        for level in range(1, self.levels):
+            scale_factor = 2 ** level
+            iterator_output = dict(
+                maxWidth=self.sizeX // scale_factor,
+                maxHeight=self.sizeY // scale_factor,
+            )
+            for frame in metadata.get('frames', [{'Index': 0}]):
+                frame_position = {
+                    k.replace('Index', '').lower(): v
+                    for k, v in frame.items()
+                    if k.replace('Index', '').lower() in self._axes
+                }
+                for tile in self.tileIterator(
+                    tile_size=tile_size,
+                    tile_overlap=dict(x=4, y=4),
+                    frame=frame['Index'],
+                    output=iterator_output,
+                    resample=False,  # TODO: incorporate resampling in core
+                ):
+                    new_tile = downsampleTileHalfRes(tile['tile'], resample_method)
+                    overlap = {k: int(v / 2) for k, v in tile['tile_overlap'].items()}
+                    new_tile = new_tile[
+                        slice(overlap['top'], new_tile.shape[0] - overlap['bottom']),
+                        slice(overlap['left'], new_tile.shape[1] - overlap['right']),
+                    ]
+
+                    x = int(
```

Comment:
You should use `(tile['x'] + tile['overlap']['left']) / 2`.

Run the samples with a tile size that is smaller than the images to make everything looks right -- the tile size includes the overlap (see https://github.com/girder/large_image/blob/master/large_image/tilesource/base.py#L2258-L2272)

Reply:
Thanks, changed in [9a74c0a](https://github.com/girder/large_image/pull/1490/commits/9a74c0ac85f7c6394c25d0a29e395f0c371f337f)